server/mongo_data_access.go

```diff
@@ -96,19 +154,45 @@ func (dal *mongoDataAccessLayer) Delete(id, resourceType string) error {
 		return convertMongoErr(err)
 	}
 
+	if dal.hasInterceptorsForVerbAndType("DELETE", resourceType) {
+		// Although this is a delete operation we need to get the resource first so we can
+		// run any interceptors on the resource before it's deleted.
+		resource, _ := dal.Get(id, resourceType)
```

Comment:
I wouldn't throw out the error (`_`) -- because if there _is_ an error (like if the resource isn't found), you probably _don't_ want to invoke the interceptor.

In fact, I wonder if that points to an issue for all of this code?  Maybe interceptors should be invoked AFTER the db operations, just to be sure they actually succeeded?  For example, even for POST and PUT -- if the insert/update fails, the interceptor probably should NOT be called.  What do you think?